java/refactoring.java/src/org/netbeans/modules/refactoring/java/plugins/RenameTransformer.java

```diff
@@ -254,7 +254,7 @@ public boolean accept(Element e, TypeMirror type) {
             String useThis = null;
             String useSuper = null;
 
-            if (elementToFind!=null && elementToFind.getKind().isField()) {
+            if (elementToFind!=null && elementToFind.getKind().isField() && tree.getKind() != Tree.Kind.MEMBER_SELECT) {
```

Comment:
Please add reasoning why you propose a change. In this case, the change looks wrong, as it contradicts the check in line 262. That branch can't be reached anymore with your change in place.

Reply:
Good point. I think the whole if statement started from line 262 can be deleted. Because this part of code (started from line 257) checks if a new name for refactored field doesn't have the same name as some local variables do. For this case it is adding an outer scope (this or class name for static fields). But I think if field is a member i.e. tree.getKind() == Tree.Kind.MEMBER_SELECT we don't need to do anything because we have a scope.

Reply:
I think I agree with @jherkel - the code on lines 262-267 does not make much sense to me. If a reference to a field is qualified, it cannot clash with a local variable (AFAIK), even if the qualifier is different from `this`. I agree we should delete the whole `if` starting at 262. Maybe instead of `tree.getKind() != Tree.Kind.MEMBER_SELECT` do `&& tree.getKind() == Tree.Kind.IDENTIFIER`, because I believe that is the only case where the clash can occur.